library/datasort.c

```diff
@@ -1687,6 +1687,8 @@ int eblob_generate_sorted_data(struct datasort_cfg *dcfg)
 		goto err_unlock_bctl;
 	}
 
+	pthread_rwlock_wrlock(&dcfg->b->iteration_lock);
+
```

Comment:
we can leave this locked if we jumt to label on 1697

Reply:
fixed it